src/main/scala/com/johnsnowlabs/nlp/util/io/ResourceHelper.scala

```diff
@@ -56,33 +56,45 @@ object ResourceHelper {
   def getSparkSessionWithS3(
       awsAccessKeyId: String,
       awsSecretAccessKey: String,
-      awsSessionToken: String): SparkSession = {
+      hadoopAwsVersion: String,
+      region: String = "us-east-1",
+      s3Impl: String = "org.apache.hadoop.fs.s3a.S3AFileSystem",
+      pathStyleAccess: Boolean = true,
+      credentialsProvider: String = "TemporaryAWSCredentialsProvider",
```

Comment:
Since we are using spark configuration, we could define the spark config in `ConfigHelper` object, and set the default value in `ConfigLoader` object.
Check `AWSGateway` class as a reference

Reply:
@danilojsl `ConfigHelper.getConfigValueOrElse` will try to load from the current spark session first, only then will it default to the provided default value in `ConfigLoader`. I'm afraid the config will never be applied to the spark session (Spark will first be created by `ResourceHelper.spark.getActiveSparkSession`). 

Should we keep the hard coded jar versions in `ConfigHelper` though?

Reply:
@DevinTDHa This is for a unit test that is ignored, it requires a manual config/set up so no worries for now. We can look into this after we merge everything into the base branch as it won't affect any functionality outside the test module.